src/builder/components/question/card/survey-question-card.component.tsx

```diff
@@ -0,0 +1,98 @@
+import { message } from 'antd';
+import React, { Component } from 'react';
+import { SurveyItem } from '../../../../common/types/survey-item.type';
+import { EditableTitle } from '../editable-title/survey-question-editable-title.component';
+import { SurveyQuestionHighlight } from '../highlight/survey-question-highlight.component';
+import { SurveyQuestion } from '../survey-questions/survey-question.component';
+import { SurveyQuestionToolbox, ToolboxActions } from '../toolbox/survey-question-toolbox.component';
+import './survey-question-card.component.less';
+
+export interface SurveyQuestionCardProps {
+    question: SurveyItem;
+    onDelete: (itemId: string) => void;
+    onUpdate: (item: SurveyItem) => void;
+}
+
+export interface SurveyQuestionCardState {
+    isToolboxVisible: boolean;
+}
+
+export class SurveyQuestionCard extends Component<SurveyQuestionCardProps, SurveyQuestionCardState> {
+    state = {
+        isToolboxVisible: false,
+    };
+
+    deleteMessage = () => {
+        message.success(`${this.props.question.description} removed successfully!`);
+    };
+
+    showToolbox = () => this.setState({ isToolboxVisible: true });
+    hideToolbox = () => this.setState({ isToolboxVisible: false });
+
+    editSurveyTitle = (newTitle: string) => {
+        const { question } = this.props;
+        question.description = newTitle;
+        this.props.onUpdate(question);
```

Comment:
For the sake of immutability, we can return a new object here:
```
editSurveyTitle = (description: string) => {
  const { question } = this.props;

  this.props.onUpdate({ 
    ...question,
    description
  });
}

It seems counterintuitive to have the `editTitle` function change the `description` though. We need to align this

Reply:
I'll introduce the immutability because that's just a good idea.

On the topic of semantics, I'm not so sure where to go from here. I named it `editableTitle` because that's what seemed appropriate at the time.

It's definitely the "question" that is asked during the surveying, so I'm not sure what to call it?

`question.question`? `question.title`? `question.name`? `question.description`? All of these are valid, but like you said, we should agree on one of them and keep it consistent

Reply:
Starting from this, which changes do you propose?
```
Survey
  SurveyItem
    Title
    Description
```

Not sure if you wanna call them `Question` instead of `SurveyItem` but that's equally good

Reply:
I think I would prefer `Question` instead. 

`SurveyItem` can be the wrapper type for all different `items` in the survey, e.g:

* Question
	* Multiple choice
	* Boolean choice
	* Open question
	* Date range
	* ...
* survey section delimiters
* ... progress indicators ... (?)
* supportive audiovisual elements
	* images
	* video with explanation about an upcoming section of the survey
	* sound clip
* interactive elements
	* embedded game (?) (often seen in university psych surveys) 
	* other interactive elements
* other elements the user may add to the survey that are not a type of question
* ...